### ISSUE TITLE
Traffic Control Inspections Monthly Metrics Report (TL)

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#1742]
## Monthly Traffic Control Inspection Metrics Report

**Purpose:** To aid in the monthly performance metrics provided to executive management 

**Primary User:** Executive Management
[TCI Reports.xlsx](https://github.com/cityofaustin/atd-data-tech/files/4846966/TCI.Reports.xlsx)

![image](https://user-images.githubusercontent.com/49930868/86034489-52caa500-ba00-11ea-86ea-465da34532bc.png)

-------------

**User Search Parameters:**
* Start Date
* End Date
* Previous Month
* Month to Date

**Query Parameters:**
* All User Parameters
* TCI Fees
* TCI Attempt Results

**Report Data (Include for each Item that meets parameters):**
* Total amount of inspection fees invoiced for the month, grouped by the inspector who created the invoice
* Total amount of inspection reports created for the month, grouped by the inspector who created the report
    * Report count = count # of attempt results on TCI process

**Misc:**
* User does not need to see details for this information, just the totals
* Consider 1st grouping to be by the inspector, then provide the totals under his/her name

*Migrated from [atd-amanda #149](https://github.com/cityofaustin/atd-amanda/issues/149)*

**Comment by atdservicebot** [comment]
*From TracyLinder on 2019-10-09*:

This request is the second part of issue #147

**Comment by atdservicebot** [comment]
*From TracyLinder on 2019-10-09*:

* Synced w/ R. Ward about this request this morning. Notes clarifying the request have been added to the description. I'll also be creating a new ticket for the 2nd report request. 
* I told Ryan I can write the queries for these reports this week, and will submit a CTM ticket to have the AMANDA CTM team do the Crystal Report work. These reports need to be available to the user in the ATD custom report menu. 
* Ryan will provide me with the 5 or 6 inspection fees that are being used.

**Comment by atdservicebot** [comment]
*From johnclary on 2019-11-21*:

@TracyLinder email Mooney about this.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan Per Paloma there are multiple TCI report requests. Please look for all TCI Requests and I will schedule a meeting with Ryan to see if there is any overlapping in information and refine the requirements.

**Comment by TracyLinder** [comment]
@Nadin-Nader will check w/ Sree for an update on this today.

**Comment by TracyLinder** [comment]
I just updated the description; this is ready for development.

**Comment by TracyLinder** [comment]
**We talked to Ryan today to demo the report in development and get clarification.** 
Notes:
- No grouping by attempt type
- Group by inspector (user)
- Ryan said include fees that have been inserted but not billed
- Query the fee stamp user to figure out which inspector inserted the fee

**Comment by TracyLinder** [comment]
[TCI Metrics Summary Report 07-13-2020.pdf](https://github.com/cityofaustin/atd-data-tech/files/4921494/TCI.Metrics.Summary.Report.07-13-2020.pdf)

Sree sent us this preview of the report. It looks great. Next we need to verify the data it's pulling is correct.

**Comment by TracyLinder** [comment]
**Status:** Incorporating the feedback received on 7/10

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder  - Completed and submitted the report on 07/24/2020

**Comment by TracyLinder** [comment]
**Status:** Currently in branch, pending Tracy's review of email sent on 7/24. Updating to sprint 30.

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader  - Waiting for feedback

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan 
New REPORT #'s 56220 56230 
Parameters: Start Date and End Date
File path to save to: \\coacd.org\dfs\Amanda\BrowserReports\DEV\ATD Reports\

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder I have deployed the report to DEV

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader - I am able to test it in DEV and ready for DEV UAT.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder I have copied this to the DEV folder path. Please log on and verify all is well and let me know the outcome.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan trying to run this one on DEV and i am getting an InternalFormatterEception error:

![image.png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/cc6a55f8-b994-482a-8d57-91b628a00e9f)

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder Re-tested with Sree and the error is gone after she fixed the parameter.

### **Approved for TEST!!🎉**

**Comment by TracyLinder** [comment]
@Sreelatha-BlazeKannan @Nadin-Nader I just looked at this report on DEV and the data is way off. I'm going to schedule time to discuss this one next week.

**Comment by TracyLinder** [comment]
Meeting scheduled for 10/13, but we'll reschedule if Sree isn't back yet.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Meeting Scheduled for 10/13/2020 to discuss in detail. however, if required we can have an internal meeting to make sure we are all on the same page.

**Comment by TracyLinder** [comment]
@Sreelatha-BlazeKannan will do research to see why old data is populating the report.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Analysis is in Progress.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder  - Sent updated reports on 10/15/2020 for Review.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan Tracy will review this and let you know. So far Version 2 looks good.

**Comment by TracyLinder** [comment]
Hi Everyone,

I prefer version 2, but I’m unable to verify the data. Can this be moved to DEV?

Thanks,
Tracy

**Comment by TracyLinder** [comment]
**For Reference, This is # 2**

![image](https://user-images.githubusercontent.com/49930868/96929311-88980b80-147f-11eb-9af6-da0a2a3c5f8b.png)

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder Version 2 is now on DEV to UAT

**Comment by TracyLinder** [comment]
### Feedback from Ryan:

Add column for folderRSN & folder type

![image.png](https://images.zenhubusercontent.com/5cca1e522b8dfc0ee14dedf5/7579dea0-08fe-4a05-9da1-a5cb5ac65a1a)

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader @TracyLinder - Acceptance Criteria to update 2 additional columns to the report; Sree will be working on this changes.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @SubhaShroff @TracyLinder  I have deployed this report 56230 to DEV an TEST and was able to run it successfully. The report has the changes needed (folder type, folder RSN). I think its ready for PROD

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Tested in DEV & TEST. Ready for UAT

**Comment by TracyLinder** [comment]
### Test Results (Failed)

**Test 1 (Failed):**
1. On DS test RSN 12384595, I added, billed, and paid fee 50333 ROW Investigation Fee (Improper Use of Device)
2. I generated TCI Metrics report for 10/1 - 10/31
3. The fee from step 1 above did not populate:

![image](https://user-images.githubusercontent.com/49930868/97497174-18cec880-1938-11eb-9c29-a00b686c63df.png)

**Test 2 (Passed):**
1. On same DS folder, RSN 12384595, I saved attempt 'Meetings' on Traffic Control Inspections Process
2. I generated TCI Metrics report for 10/1 - 10/31
3. The deficiency populated, as well as the fee from test 1

![image](https://user-images.githubusercontent.com/49930868/97497524-9eeb0f00-1938-11eb-8ebf-2c13538a9cc0.png)

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder -  YES Tracy, that's how the join was written that is only when an attempt made on TCI Inspection process it will look for the fee. If you want to report irrespective of the report count then I will have to update the query.

**Comment by TracyLinder** [comment]
@Sreelatha-BlazeKannan thank you for the clarification. Please update the query to include the fee irrespective of the report count. Are you able to join the fee & report count when both exist?

**Comment by TracyLinder** [comment]
**Report Update Template:**

![image](https://user-images.githubusercontent.com/49930868/97635383-47ff3b80-1a05-11eb-84d4-8a97d9b8cd25.png)

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Report work is in progress

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Report Format ready for review on 11/2/2020

**Comment by TracyLinder** [comment]
@Sreelatha-BlazeKannan the report permit looks great. Please add a section on the fee summary page that totals each fee type, irrespective of the stamp user/inspector.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder
#1742 - Completed and submitted the  report for review. on 11/3/2020 and waiting for feedback.

**Comment by TracyLinder** [comment]
### Approved to Promote to Dev/Test
@Sreelatha-BlazeKannan the format looks good!

![image](https://user-images.githubusercontent.com/49930868/98174869-2dbcd600-1ebb-11eb-8266-53b672cdd8b7.png)
![image](https://user-images.githubusercontent.com/49930868/98174895-36ada780-1ebb-11eb-9755-09e6ad1aece6.png)

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder I have deployed this to TEST and PROD and is ready for UAT.

## **Status:**
Ready for UAT on TEST

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder @Nadin-Nader -  Waiting for feedback

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder Looks like its pulling partial information
Ran it on TEST
Page 1:
![image.png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/0ca83a3f-4c90-498f-a0fe-872952f238e7)

Page 2:
![image.png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/087abc87-5bfe-4ccc-8c78-728431eb108d)

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder @Nadin-Nader  - Due to the data clean up in DEV & TEST for fees  the columns are showing empty. I did analysis and added few fees and ran the report and it is working  as expected. 
This is from TEST - 

![image](https://user-images.githubusercontent.com/44907715/99534929-ccdad680-296d-11eb-9772-d79de2beb820.png)


![image](https://user-images.githubusercontent.com/44907715/99534999-e8de7800-296d-11eb-836a-8a0629d288a0.png)

![image](https://user-images.githubusercontent.com/44907715/99535093-0875a080-296e-11eb-9b81-6d927e548cb2.png)

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder I have ran this on TEST and it works as expected:

Page 1:

![image.png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/1e0b7047-2372-4711-adc2-a1b52721af51)

Page 2:

![image.png](https://images.zenhubusercontent.com/5e8dfbfabf171441c3a5684e/23cbc6e3-18d5-4313-839c-53dc4e59e56e)

## **Status:**

Ready for Stake holder approval to move to PROD

**Comment by TracyLinder** [comment]
### UAT Results (PASSED)

### APPROVED FOR PROD 🎉

**Comment by Nadin-Nader** [comment]
@TracyLinder @Sreelatha-BlazeKannan Sent this over to the CTM Team to deploy to PROD under TASK **SCTASK0105721**

## **Status:**
Waiting on CTM to deploy to PROD

**Comment by Nadin-Nader** [comment]
This has been deployed to PROD. Closing
